include/caffe/util/gpu_memory.hpp

```diff
@@ -0,0 +1,121 @@
+#ifndef CAFFE_UTIL_GPU_MEMORY_HPP_
+#define CAFFE_UTIL_GPU_MEMORY_HPP_
+
+#include <vector>
+#include "caffe/common.hpp"
+
+namespace caffe {
+
+class gpu_memory {
+ public:
+  enum PoolMode {
+    NoPool,     // Straight CUDA malllc/free. May be very expensive
```

Comment:
Two spaces before comments, unless they're part of a multi-line block. You may also wish to spell `malloc` correctly.